packages/app/src/cli/services/app/config/link.ts

```diff
@@ -60,10 +60,10 @@ export default async function link(options: LinkOptions, shouldRenderSuccess = t
   return configuration
 }
 
-async function selectRemoteApp(options: LinkOptions, developerPlatformClient: DeveloperPlatformClient) {
+async function selectRemoteApp(options: LinkOptions) {
```

Comment:
What about something like:

```suggestion
async function selectRemoteApp(options: LinkOptions & Required<Pick<LinkOptions, "developerPlatformClient">>) {
```

Then the `!` below is unnecessary.

Alternatively, we could consider requiring `developerPlatformClient` in `LinkOptions` and fixing up any places where it isn't passed in.